src/audio_unit/looper.rs

```diff
@@ -44,20 +47,39 @@ impl Looper {
         }
     }
 
-    fn process_message(&mut self, _: Message) {
-        self.state = match self.state {
-            Off => Recording { position: 0 },
-            Recording { position } => Playing {
-                position: 0,
-                total: position,
-            },
-            Playing {
-                position: _,
-                total: _,
-            } => Off,
-        };
+    fn process_message(&mut self, message: Message) {
+        match message {
+            Message::Toggle => {
+                self.state = match self.state {
+                    Off => Recording { position: 0 },
+                    Recording { position } => Playing {
+                        position: 0,
+                        total: position,
+                    },
+                    Playing {
+                        position: _,
+                        total: _,
+                    } => Off,
+                    PlayingAwaitingOverdub {
+                        position: _,
+                        total: _,
+                    } => Off,
+                    Overdubbing {
+                        position: _,
+                        total: _,
+                    } => Off,
```

Comment:
These 3 last cases could all be matched with a single `_`:

```rust
_ => Off
```

(Although that is a bit less explicit.)

Reply:
I like that, updated to just `_ => Off`